src/activity_handlers.cpp

```diff
@@ -689,9 +700,19 @@ void activity_handlers::fill_liquid_do_turn( player_activity *act_, player *p )
             liquid.charges = item::INFINITE_CHARGES;
             break;
         case LST_MAP_ITEM:
-            if( static_cast<size_t>( act.values.at( 1 ) ) >= source_stack.size() ) {
+            if( static_cast<size_t>( act.values.at( 1 ) ) >= source_stack.size() ){
                 throw std::runtime_error( "could not find source item on ground for liquid transfer" );
             }
+            break;
+        case LST_MONSTER:
+        Creature *c = g->critter_at(source_pos);
+        source_mon = dynamic_cast<monster *>(c);
+        if( source_mon == nullptr ){
```

Comment:
With knocktwice's suggestion I realize that it is problem of the algorithm to set the effects. Working on it.